### PR TITLE
Task 1 from networking

### PR DESCRIPTION
**Pull request by qnislavnachev**

### CODE REVIEW

File: src/task1/DownloadAgent.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+public class DownloadAgent {
+
+    public File downloadFile(String link, String fileName, String extension) {
+        File file = null;
+        try {
+            final byte[] data = new byte[1024];
+            file = new File(fileName + extension);
+            BufferedInputStream reader = new BufferedInputStream(new URL(link).openStream());
+            FileOutputStream writer = new FileOutputStream(file);
```
Comment (tsony-tsonev) [inline src/task1/DownloadAgent.java]: you should define the reader and the writer outside of the try so you can close the in a finally method not inside the try block

File: test/agentTest/DownloadAgentTest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import static org.junit.Assert.*;
+
+public class DownloadAgentTest {
+
+    @Test
+    public void downloadedFileSize() throws Exception {
+        DownloadAgent agent = new DownloadAgent();
+        URL url = new URL("https://i.ytimg.com/vi/9Nwn-TZfFUI/maxresdefault.jpg");
+        File file = agent.downloadFile("https://i.ytimg.com/vi/9Nwn-TZfFUI/maxresdefault.jpg", "file", ".jpg");
+        long expected = url.openConnection().getContentLength();
```
Comment (tsony-tsonev) [inline test/agentTest/DownloadAgentTest.java]: use local file not one from the internet

**Comment by tsony-tsonev** [comment]
Good job btw

File: src/task1/DownloadAgent.java (hunk 1 above)
Comment (qnislavnachev) [inline src/task1/DownloadAgent.java]: fixed

File: test/agentTest/DownloadAgentTest.java (hunk 2 above)
Comment (qnislavnachev) [inline test/agentTest/DownloadAgentTest.java]: fixed

**Comment by qnislavnachev** [comment]
thank you

File: src/task1/DownloadAgent.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+public class DownloadAgent {
+
+    public void downloadFile(BufferedInputStream reader, FileOutputStream writer) throws IOException {
+        final byte[] data = new byte[1024];
+        int count;
+        while ((count = reader.read(data, 0, 1024)) != -1) {
+            writer.write(data, 0, count);
+        }
+        reader.close();
+        writer.close();
```
Comment (tsony-tsonev) [inline src/task1/DownloadAgent.java]: I meant that you should have defined the reader and the writer before the try block(not outside of the method) so you can close them in the finally section, this way if something goes wrong your files wont stay opened forever.

Comment (qnislavnachev) [inline src/task1/DownloadAgent.java]: oh i am sorry i didnt understand you i will fix it

Comment (qnislavnachev) [inline src/task1/DownloadAgent.java]: I think now its fixed

File: test/agentTest/DownloadAgentTest.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+public class DownloadAgentTest {
+
+    @Test
+    public void downloadedFileSize() throws Exception {
+        DownloadAgent agent = new DownloadAgent();
+        File file = new File("pic1.jpg");
+        URL url = file.toURI().toURL();
+        String link = url.toString();
+        File downloaded = agent.downloadFile(link, "file", ".jpg");
```
Comment (tsony-tsonev) [inline test/agentTest/DownloadAgentTest.java]: you dont need url you can pass the file path as string then you dont need a connection to get the length you can get it from the file

**Comment by tsony-tsonev** [comment]
LGTM when you fix the last comment

**Pull request merged**